honeycombio/honeycomb-network-agent
# feat: Add more Kubernetes metadata

## Which problem is this PR solving?
Adds more Kubernetes metadata to events. Followed the conventions listed in OTel: https://opentelemetry.io/docs/specs/otel/resource/semantic_conventions/k8s/

- Closes #49 

## Short description of the changes
Added:
- source / dest UIDs
- namespace name
- service name
- node name / UID
- container name

TODO:
- ~[ ] Figure out how to get cluster name~
- [x] Check that service name is working correctly
- [x] Use semconv package for field names
- [x] Normalize source / dest IP names
- ~[ ] Write unit tests for k8s metadata transformation functions~



## How to verify that this has the expected result
- Run in a k8s cluster and send data to Honeycomb
- Make sure to run greeting service to see more interesting data
![Screenshot 2023-08-01 at 5 56 30 PM](https://github.com/honeycombio/honeycomb-ebpf-agent/assets/8810222/3c373f01-d1e7-47c6-ab2e-35f9b25218ae)

Data being emitted:


![Screenshot 2023-08-03 at 3 11 01 PM](https://github.com/honeycombio/honeycomb-ebpf-agent/assets/8810222/76081824-cf91-4303-a165-b9d168b10f18)
